src/workerd/io/worker.c++

```diff
@@ -758,26 +759,22 @@ struct Worker::Script::Impl {
   // If set, then any attempt to use this script shall throw this exception.
   kj::Maybe<kj::Exception> permanentException;
 
-  void configureDynamicImports(jsg::Lock& js, jsg::ModuleRegistry& modules) {
-    struct DynamicImportResult {
-      jsg::Value value;
-      bool isException = false;
-    };
-
-    using Handler = kj::Function<jsg::Value()>;
-
-    static auto constexpr handleDynamicImport =
-        [](kj::Own<const Worker> worker,
-           Handler handler,
-           kj::Maybe<jsg::Ref<jsg::AsyncContextFrame>> asyncContext) ->
-               kj::Promise<DynamicImportResult> {
-      co_await kj::evalLater([] {});
-      auto asyncLock = co_await worker->takeAsyncLockWithoutRequest(nullptr);
-
-      Worker::Lock lock(*worker, asyncLock);
+  struct DynamicImportResult {
+    jsg::Value value;
+    bool isException = false;
+  };
+  using DynamicImportHandler = kj::Function<jsg::Value()>;
+
+  static DynamicImportResult getDynamicImportResult(
+      Worker::AsyncLock& asyncLock,
+      auto& worker,
```

Comment:
Does this `auto&` always resolve to `const Worker&`? If so we should really just write out the type. `auto`-typed function arguments turn the function into a template which, among other things, breaks IDE auto-completion, so should generally be avoided.